lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -70,106 +70,93 @@ String FunctionalChaosSobolIndices::__str__(const String & /*offset*/) const
   const UnsignedInteger inputDimension = functionalChaosResult_.getDistribution().getDimension();
   const UnsignedInteger outputDimension = functionalChaosResult_.getMetaModel().getOutputDimension();
   OSS oss;
+  oss << FunctionalChaosSobolIndices::GetClassName() << "\n";
 
   const Indices indices(functionalChaosResult_.getIndices());
   const Sample coefficients(functionalChaosResult_.getCoefficients());
   const UnsignedInteger basisSize = indices.getSize();
+  EnumerateFunction enumerateFunction(functionalChaosResult_.getOrthogonalBasis().getEnumerateFunction());
 
-  // compute the variance
-  Point variance(outputDimension);
+  // compute the mean, the variance, the standard deviation
   Point mean(outputDimension);
+  Point variance(outputDimension);
+  Point stdDev(outputDimension);
   for (UnsignedInteger i = 0; i < outputDimension; ++i)
   {
     for (UnsignedInteger k = 0; k < basisSize; ++ k)
-      // Take into account only non-zero indices as the null index is the mean of the vector
-      if (indices[k] > 0)
-        variance[i] += coefficients(k, i) * coefficients(k, i);
-      else
+    {
+      // Take into account only non-zero total degree indices as the
```

Comment:
This does not seem right to me, because `FunctionalChaosRandomVector.getMean()` and `FunctionalChaosRandomVector.getCovariance()` already implement these equations. Why not using them?

Reply:
@jschueller Do you have any will not to use FunctionalChaosRandomVector within FunctionalChaosSobolIndices, or is this just a code that was developed before FunctionalChaosRandomVector.getMean() and getCovariance() exist?

Reply:
feel free to factorize then

Reply:
Ok. Fixed at f3bb8b618.